[18.0][MIG] product_pricelist_direct_print

bosd opened the pull request:
depends on: #1864

carlosdauden committed:
[9.0][ADD] product_pricelist_direct_print (#223)

carlosdauden committed:
[IMP] product_pricelist_direct_print: Add order option + change widget for pricelist (#290)

chienandalu committed:
[MIG] product_pricelist_direct_print: Migration to 11.0

ernestotejeda committed:
product_pricelist_direct_print: extend module

Print price list for future date
Print price list for N last ordered products
Send report by email to each partner selected

sergio-teruel committed:
[11.0][FIX] product_pricelist_direct_print: Remove duplicate default code

alexey-pelykh committed:
[MIG] product_pricelist_direct_print: Migration to 12.0

sergio-teruel committed:
[IMP] product_pricelist_direct_print: black, isort

sergio-teruel committed:
[MIG] product_pricelist_direct_print: Migration to v13.0

alexey-pelykh committed:
[UPD] Brainbean Apps => CorporateHub

joao-p-marques committed:
[MIG] product_pricelist_direct_print: Migration to 14.0

TT26448

sergio-teruel committed:
[IMP] product_pricelist_direct_print: Improve inheritability
TT27367

sergio-teruel committed:
[IMP] product_pricelist_direct_print: Add summary on report and wizard.
Ported from commit b317151ff3eb53b962d54f576f8a18a1c25b4e68
TT27367

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: forward port of xlsx and just defined products

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: Date adjusted to the selected language

CarlosRoca13 committed:
[FIX] product_pricelist_direct_print: Error when date is not defined

When we check the price of any product, the date that we pass on the context has to be a Datetime value or None. With this change we avoid to pass a False.

Here you have the reference to the code that check that https://github.com/OCA/OCB/blob/13.0/addons/product/models/product.py#L599-L600

TT29935

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: Possibility of break by parent categories

carlosdauden committed:
[FIX] product_pricelist_direct_print: Revert last changes

carlosdauden committed:
[IMP] product_pricelist_direct_print: Add max category level to group products
TT30596

chienandalu committed:
[FIX] product_pricelist_direct_print: company in report

In a multicompany environment we need to contextualize the report
company. Otherwise, the pricelists will be printed with the header and
data of the current user's default company, which can be incoherent.

TT32544

chienandalu committed:
[FIX] product_pricelist_direct_print: fix date context

This fix a compatibility issue with product_pricelist_supplierinfo if
the wizard date isn't set.

TT32729

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: Improve inheritance at excel + add new fields + Adjust cherry-picks to v14 + Give formats to add_extra_info method

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: Improve inheritance on get_last_ordered_products_to_print

CarlosRoca13 committed:
[MIG] product_pricelist_direct_print: Migration to 15.0

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: Allow select the language when printing

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-15.0/product-attribute-15.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_pricelist_direct_print/

legalsylvain committed:
[MIG] product_pricelist_direct_print 16.0 (from 15.0)

legalsylvain committed:
[IMP] product_pricelist_direct_print : display product UoM + display pricelist prices vat excl or vat incl + group by any product field

legalsylvain committed:
[REF] product_pricelist_direct_print: Several refactorings

- Improve wizard display, grouping fields by context
- Harmonize fields names : Use only 'positive' option 'show' and remove 'hide' option
- Set date as required, simplifying algorithm
- Split product_pricelist_direct_print into product_pricelist_direct_print and product_pricelist_direct_print_xlsx

CarlosRoca13 committed:
[FIX] product_pricelist_direct_print: Migrate changes of mail template

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (65 of 65 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/es/

dcorio committed:
Added translation using Weblate (Italian)

Currently translated at 84.6% (55 of 65 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 84.6% (55 of 65 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/it/

pasculorente committed:
[16.0] [FIX] product_pricelist_direct_print: add missing dependency in compute

The _compute_product_price method is missing the depends line, causing it not to be recomputed during the same report, thus all products result in the same price

dcorio committed:
Translated using Weblate (Italian)

Currently translated at 84.6% (55 of 65 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/it/

josep-tecnativa committed:
[IMP] product_pricelist_direct_print: test performance improvement
- Include context keys for avoiding mail operations overhead.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (65 of 65 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/it/

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: Allow printing the products sold from x date

Commit:
Added translation using Weblate (Turkish)

suker committed:
[FIX] product_pricelist_direct_print: bugfix
 Blank product_attribute_values on changing show_variants

sergio-teruel committed:
[IMP] product_pricelist_direct_print: Allow define the category level to print in report
TT49004

trisdoan committed:
[FIX] product_pricelist_direct_print: fix bug 'product.pricelist.print' object has no attribute 'show_attribute_values'

CarlosRoca13 committed:
[FIX] product_pricelist_direct_print: Error when display wizard without administration permision

Steps to reproduce the problem:

1. Log in with a user without Administration permision
2. Open Pricelist Print wizard

An error about permisions is desplayed

legalsylvain committed:
[FIX] product_pricelist_direct_print : Prevent access error if a user that doesn't belong to 'Administration/Access Rights' try to print pricelists. Fix : https://github.com/OCA/product-attribute/pull/1655

chaule97 committed:
[IMP] product_pricelist_direct_print: printing a pricelist and selecting an product category, include its child categories.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (66 of 66 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/it/

therbin200 committed:
[MIG] product_pricelist_direct_print: Migration to 17.0

bosd committed:
[IMP] product_pricelist_direct_print: pre-commit auto fixes

[IMP] product_pricelist_direct_print: pre-commit execution

bosd committed:
[MIG] product_pricelist_direct_print: Migration to 18.0

- Standard procedure.
- [IMP] Add advanced print button
- [IMP] Add option to show Sales Description
- [FIX] typo product_pricelist_print_view
- [IMP]Improve string Show only defined products
- [IMP] Pretty Filename: Use pricelist- name instead of report.pdf in emails.

bosd commented:
This option is not working yet. It adds all items from the db.
![image](https://github.com/user-attachments/assets/9ea986cd-916b-4afc-bcc9-63b575030f36)


Edit: Fixed now :green_heart:

Dranyel-Bosd reviewed (approved):
LGTM :1st_place_medal:

CarlosRoca13 committed:
[IMP] product_pricelist_direct_print: Add decimal precision to price fields

With this change we are sure that the prices shown on the report has the correct precision.

bosd commented:
@legalsylvain Can you please review?

bosd commented:
@CarlosRoca13 Can you please review this one?

CarlosRoca13 commented:
I'm not sure this reference really convinces me; it could be confusing. Couldn't we avoid adding a reference when there isn't one?
![image](https://github.com/user-attachments/assets/9cd401e2-5700-4ce1-86c6-ba82df93457a)

bosd commented:
@CarlosRoca13 I'm sorry but I don't understand your point.
The reference has not been changed by this pr.
When there is no reference given. link in example below for the `Office Combo` it will not be shown.
![image](https://github.com/user-attachments/assets/cfc82640-2b1b-42e0-bc23-7a4592b3d45b)

CarlosRoca13 commented:
It is happening when the option "Show variants" is selected:
![image](https://github.com/user-attachments/assets/7f301832-6090-43ac-90f3-9068ad62f965)

![image](https://github.com/user-attachments/assets/997662b6-249f-4ef1-a6c9-fdc43c287f69)

bosd commented:
Yes, I got that. When you set an default_code on the product_template and generate the pricelist without the variants option. Then the default_code of the product_template is shown on the pricelist. (If the field is avalable)
When you check the box `Show variants`  then the report is listing the `default_code` set on all the product variants.

When there are multiple product.products the field is not available.
eg. customizable desk:
![image](https://github.com/user-attachments/assets/b9ed5ecc-5df5-4282-85b2-dbb7af538a13)

With show variants:
![image](https://github.com/user-attachments/assets/4b76ca75-4fa9-444b-a396-d4b551b0a813)

That is exactly what is supposed to be happening. right?

CarlosRoca13 commented:
> That is exactly what is supposed to be happening. right?

I believe so, or at least it should appear as it did in previous versions, since that reference that shows up when it's not set is a bit annoying to me...

Example from previous versions:
![image](https://github.com/user-attachments/assets/a610796a-78a1-47be-a890-d6ebcca5cd3a)

victoralmau commented:
In addition to what my colleague comments on https://github.com/OCA/product-attribute/pull/1860#issuecomment-2808825321

Can you unify all commits in the migration commit?
<img width="769" height="517" alt="ejemplo" src="https://github.com/user-attachments/assets/7d69f099-87f6-4cda-a569-ecf3f7c129fd" />

pedrobaeza commented:
/ocabot migration product_pricelist_direct_print

pedrobaeza reviewed (approved):
I have rebased + squash the PR, reducing significantly the number of commits, so we can proceed with the merging:

/ocabot merge nobump
